src/utils.jl

```diff
@@ -56,6 +52,40 @@ function GradIntervalEval_test(nlp::AbstractNLPModel, FPList::AbstractArray)
   end
 end
 
+"""
+    ObjTypeStableTest(nlp::AbstractNLPModel, FPList::AbstractArray)
+
+Tests if objective evaluation of `nlp` is type stable for FP format in FPList
+"""
+function ObjTypeStableTest(nlp::AbstractNLPModel, FPList::AbstractArray)
+  for fp in FPList
+    intype = fp
+    x0 = fp.(nlp.meta.x0)
+    output = obj(nlp, x0)
```

Comment:
This might need to be in a try/catch, because `obj(nlp, x0)` might return an error if the type of `x0` is different than `output`.

Reply:
I have a test for that but I only get the expected warning, no error:
https://github.com/JuliaSmoothOptimizers/MultiPrecisionR2/pull/64/files#diff-d9bc62629bc13a1ecf114a6a9360345e8366955289ec8036ab2e556d811fc478R90

Reply:
This issue is somewhat related: https://github.com/JuliaSmoothOptimizers/MultiPrecisionR2/issues/62

Reply:
Oki